apis/identity/v1beta1/iamrolepolicyattachment_types.go

```diff
@@ -45,7 +42,7 @@ func (v *IAMRoleNameReferencerForIAMRolePolicyAttachment) Assign(res resource.Ca
 		return errors.New(errResourceIsNotIAMRolePolicyAttachment)
 	}
 
-	p.Spec.RoleName = value
+	p.Spec.ForProvider.RoleName = value
 	return nil
 }
 
```

Comment:
It wouldn't let me comment directly on these lines, but it looks like `RoleName` and and `RoleNameRef` are both optional fields. Would you mind updating them as follows:

```
        // RoleName presents the name of the IAM role.
        // +optional
	RoleName *string `json:"roleName,omitempty"`

	// RoleNameRef references to an IAMRole to retrieve its Name
        // +optional
	RoleNameRef *IAMRoleNameReferencerForIAMRolePolicyAttachment `json:"roleNameRef,omitempty" resource:"attributereferencer"`
```

Reply:
@sahil-lakhwani per your message in Slack, I think it is fine to leave `RoleName` as-is (since it is "required" for creation of the external resource but we do not validate it on object creation), but we should add an `// +optional` tag to `RoleNameRef` :) Thank you for noticing this!

Ref: https://github.com/crossplane/stack-gcp/blob/8ece46f91e44e4139ba92f848be687b970acf214/apis/container/v1alpha1/nodepool_types.go#L125